### PR TITLE
Add unique ids for each block

### PR DESCRIPTION
**Pull request by christoph-kluge**
I'm adressing #873 with this PR.

Tested Features:

* Existing data: If your previous block does not have an "id" property, it will create a new unique id
* Existing data: If your previous block does have an "id" property, it will keep the existing unique id
* Existing data: If a block gets changed the "id" property stays the same
* Insert New Block: It will create a new unique id
* Copy & Paste blocks: It will create a new unique ids per copied block
* Move Blocks (Up/Down): It will keep the existing unique id

Important Note:
* If you "transform/transition" an existing block to another block it will receive a NEW unique id

### CODE REVIEW

**Review by dsaralaya — APPROVED** [review]

**Comment by sis-dk** [comment]
Does this handle creating a new id when copying a block and pasting it as a new block? Initialise empty editor, type a para block, CMD+A (selects all the blocks. In our case the para block), CMD+V. Would the newly pasted block have a new id?

**Comment by christoph-kluge** [comment]
@sis-dk thanks for spotting the copy & paste issue. I have adressed this in my latest commit

**Comment by mqtik** [comment]
Why don't use shortid for id generation?
It's what I'm about to use in a module I'm working on (native side), but this one uses shortid to generate ID's.

Just want to keep consistent.

**Comment by christoph-kluge** [comment]
I have used this feature on a custom build since 2.15.2 with UUIDs. Now I wanted to update for latest >2.18+ version and simply ported this feature with UUIDs. I think we could also add some custom configuration/callback to generate the IDs in a different way. Since One editor will not have quadrillions of entries I could imageine that shortids are pretty valid too 👍 

Maybe the maintainers (@neSpecc, @khaydarov, @gohabereg) could give a statement about this. I'm fine to add some custom configuration/callback to support custom-ids instead of hardcoded uuids.

File: src/components/modules/renderer.ts [hunk 1]
```diff
@@ -63,12 +65,14 @@ export default class Renderer extends Module {
    */
   public async insertBlock(item: OutputBlockData): Promise<void> {
     const { Tools, BlockManager } = this.Editor;
+    const id = item.id || _.generateUuidv4();
```
Comment (hata6502) [inline src/components/modules/renderer.ts:68]:
```suggestion
    const id = item.id;
```

When `const id` is undefined, a new ID is generated in `BlockManager.insert()`.

**Review by hata6502 — CHANGES_REQUESTED** [review]

**Comment by ChasLui** [comment]
I need this feature pls merge @christoph-kluge 🙏

**Comment by neSpecc** [comment]
Resolved by #1667

**Pull request closed without merging**